### ISSUE TITLE
0.12.7 tests fail with python 2.7 in test_server/tearDown

### ISSUE THREAD

**Issue by felixonmars** [bottlepy/bottle#618]
The full error message:

```
======================================================================
ERROR: test_simple (test_server.TestServer)
Test a simple static page with this server adapter.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-bottle/src/python2-bottle/test/test_server.py", line 77, in tearDown
    raise AssertionError(line.strip().decode('utf8'))
AssertionError: AttributeError: 'NoneType' object has no attribute 'split'

----------------------------------------------------------------------
Ran 328 tests in 6.375s

FAILED (errors=1)
```

**Comment by simevo** [comment]
the issues with the tests are a nuisance for debian packaging, see https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=753220 and more recently https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=834918

I can run the entire test suite with pytest like this:
```
PYTHONPATH=. py.test -vvv
```

turns out the test "driver" testall.py is not required and pytest makes it easier to launch individual tests (as in `PYTHONPATH=. py.test -vvv test/test_sendfile.py::TestSendFile::test_invalid`) or to skip some (as in `PYTHONPATH=. py.test -vvv -k 'not test_sendfile and not test_server'`)

anyway today on my test machine (debian amd64 stretch lxc container with python 2.7) the following tests fail:
```
test/test_sendfile.py::TestSendFile::test_invalid FAILED
test/test_server.py::TestServerAdapter_gunicorn::test_simple FAILED
```

I was surprised to see test_sendfile failing: this github issue, the related #891 and the debian bugs above all reference test_server

does this come as a surprise to you ?

**Comment by defnull** [comment]
Yes, I know that the server tests are unstable (and should probably be disabled in debian builds because they are more integration than unit tests) but ``TestSendFile::test_invalid`` should not fail. Do you have any details what exactly failed or can you reproduce the error?

**Comment by simevo** [comment]
the `TestSendFile::test_invalid` failed with this error:
```>           self.assertEqual(403, static_file(fn, root='/').status_code)
E           AssertionError: 403 != 200```

it happens only when the test suite is run as root: chmod does set to 000 the permissions, but root can read the file nonetheless

**Comment by defnull** [comment]
Okay, running tests as root is something I did not consider. In docker/container test environments this might actually be the normal case.
